### ISSUE TITLE
Crawler: Setup: Separate tag name question from setup and move to crawle

### ISSUE THREAD

**Issue by BioPhoton** [timdeschryver/find-deprecations#33]
IMHO the targe branch/tag should not be asked first. AFAICT it is asked for every run (besides if we pass it directly)?

I would suggest moving this question into the crawl phase where we generate the war json for a specific branch/tag.
This would also improve the DX as we would have the setup first one single time and then always start with the

A suggestion for a one time setup process could be:
- √ Pick a ts config file to target files to crawl
  - Do you want to include or exclude other files?
  - √ Enter a glob pattern to include files, leave blank to skip.
  - √ Enter a glob pattern to exclude files, leave blank to skip.
- √ What's the deprecation keyword to look for?
- √ What's the deprecation link to the docs?
- √ What's the output directory?

**Comment by timdeschryver** [comment]
I agree, I wanted to discuss this with you but it seems like we're thinking the same thing.
It basically is a left-over from the first versions.

I also thought of removing the `checkout` step.
As discussed previously, the crawling phase should run on every commit.
I see no need anymore to switch between branches.
Thoughts?

**Comment by BioPhoton** [comment]
I think switching is important to detect in which branch a deprecation got introduced?
like i want to crawl from tag 5.6.6 to 7.4 and list all deprecations in all tags. 

Deprecations get introduced in a version and break in a version.

**Comment by timdeschryver** [comment]
This is something that isn't handled currently.
It only crawls one branch.
It doesn't go from branch X to branch Y.

**Comment by BioPhoton** [comment]
We should specify in #26 when and how the raw JSON is committed and if there are one or multiple files. 

I think about the question "I'm on v4 and want to migrate to v6 what to do?"
And I would need a set of steps to migrate to v5 and one to v6.
